### DISCUSSION TITLE
Question about queryOption pattern

### DISCUSSION THREAD

**Discussion by wildfrontend** [TanStack/query#7591] [Q&A]
when i use prefetch ,the queryFn would be fetch api from server side,

but almost time for loading balance we would be use interal  api url to fetch data

so when we have to clien side queryFn and server side queryFn 

but in queryOption example , it is use exteranl api url to fetch data , is it a good design?

**Comment by TkDodo** [comment]
you can either provide a different queryFn on the server by overwriting the default one, or you can do a check inside the queryFn if you are on the server or the client and behave differently then (this is what trpc is doing for example)
